app/src/sharedTest/java/org/oppia/android/app/profileprogress/ProfileProgressFragmentTest.kt

```diff
@@ -360,25 +359,19 @@ class ProfileProgressFragmentTest {
   @Test
   fun testProfileProgressFragment_imageSelectAvatar_configChange_checkGalleryIntent() {
     val expectedIntent: Matcher<Intent> = allOf(
-      hasAction(Intent.ACTION_PICK),
-      hasData(MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+      hasAction(Intent.ACTION_GET_CONTENT),
+      hasType("image/*")
     )
     val activityResult = createGalleryPickActivityResultStub()
     intending(expectedIntent).respondWith(activityResult)
     launch<ProfileProgressActivity>(createProfileProgressActivityIntent(internalProfileId)).use {
+      onView(isRoot()).perform(orientationLandscape())
       testCoroutineDispatchers.runCurrent()
       clickProfileProgressItem(itemPosition = 0, targetViewId = R.id.profile_edit_image)
       verifyTextInDialog(context.getString(R.string.profile_progress_edit_dialog_title))
       onView(withText(R.string.profile_picture_edit_alert_dialog_choose_from_library))
         .perform(click())
-      testCoroutineDispatchers.runCurrent()
       intended(expectedIntent)
-      onView(isRoot()).perform(orientationLandscape())
-      testCoroutineDispatchers.runCurrent()
-      intended(expectedIntent)
-      clickProfileProgressItem(itemPosition = 0, targetViewId = R.id.profile_edit_image)
-      // The dialog should still be open after a configuration change.
-      verifyTextInDialog(context.getString(R.string.profile_progress_edit_dialog_title))
```

Comment:
You have a good point in refactoring this test, but I think some of the removed assertions are still needed. I'd like to propose spiltting this into 2 seperate tests instead.

1. `testProfileProgressFragment_imageSelectAvatar_configChange_checkGalleryIntent`
This is the same as your current changes, testing that the gallery intent is still available in landscape.

2. Add new test `testFragment_imageSelectAvatar_configChange_profilePictureDialogIsVisible()`
This test should contain the lines of code that you have removed, to verify that
- user clicked on the profile avatar
- the edit profile picture dialog is visible
- user changed orientation to landscape
- the edit profile picture dialog is still visible

You can try out these steps manually in order to better understand the behaviour I am describing.

Reply:
Are you trying to say that someone opens the profile picture dialog and instead of dismissing the dialog simply rotates the screen and this time the dialog should remain visible to the user?

Reply:
Yes, see:
[device-2023-08-09-142348.webm](https://github.com/oppia/oppia-android/assets/59600948/3b8ecd53-2083-42b5-87c6-31e4383ca084)

Reply:
I'm also trying to recording my screen to show you, lol. 
I understood.